packages/pretty-format/src/index.js

```diff
@@ -57,15 +59,15 @@ function printFunction(val, printFunctionName) {
   }
 }
 
-function printSymbol(val) {
+function printSymbol(val: Symbol): string {
   return symbolToString.call(val).replace(SYMBOL_REGEXP, 'Symbol($1)');
 }
 
-function printError(val) {
+function printError(val: Error): string {
   return '[' + errorToString.call(val) + ']';
 }
 
-function printBasicValue(val, printFunctionName, escapeRegex) {
+function printBasicValue(val: any, printFunctionName: boolean, escapeRegex: boolean): StringOrNull {
```

Comment:
Right, you explain `mixed` vs `any` clearly. So, for code that’s carefully written, rarely edited, and perf critical, we might leave well enough alone with `any` in this PR?

For context, Jest had a regression in a serializer plugin recently, fixed in #2943, which Flow type checking would have prevented. Follow up includes adding enough checking to `pretty-format` to protect against, for example, editing error in args or undefined return value if a `printAmazingFeatureInES20xx` function is added.